### PR TITLE
serialize model equivalencies

### PR DESCRIPTION
**Pull request by nden**
This is an implementation of the suggested API change in #8059.
The interface can be improved by not requiring the equivalencies tuple to include empty args and kwargs, I just wanted to show how it looks. Any thoughts?

@Cadair @perrygreenfield @astrofrog

### CODE REVIEW

File: astropy/io/misc/asdf/tags/unit/unit.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class UnitType(AstropyAsdfType):
 
     @classmethod
     def to_tree(cls, node, ctx):
+        print('node', node)
```
Comment (drdavella) [inline astropy/io/misc/asdf/tags/unit/unit.py]: Looks like a leftover.

**Comment by bsipocz** [comment]
This will need a changelog entry. Could you please either open a PR to add one, or suggest the text here (so I can copy it over next time I run the consistency scripts).

**Comment by nden** [comment]
I've looked at this before because it's a strange case. This PR was never merged, not using the `merge` button anyway. If you look at the code in the repository it's different than what is in this PR. It was superseded by #8252. Files from this branch were copied to the #8252 branch and modified after the reviews. If I have to guess what happened - git kept track of the files in the new branch and marked this PR as merged when the other one was merged. The really weird part is that the other branch was in Stuart's repo and I was pushing to it. Of course I may be wrong. The good part in all this is that the code in this PR is not in the repository.

**Comment by bsipocz** [comment]
Hmm, this is super weird, and following the merge commit it indeed takes one to @Cadair's fork and has no trace in master. 
The scripts nevertheless are picking it up as a merged PR. I've added a "no-changelog-entry-needed", but if you think it add any useful information, we can list this number along #8252, too.

**Comment by nden** [comment]
I think `no-changelog-entry-needed` label is fine. Thanks for looking into this.

**Pull request merged**